### ISSUE TITLE
New text dataset

### ISSUE THREAD

**Issue by iuria21** [renmengye/few-shot-ssl-public#6]
Hi, thanks for your work. This is a question, not an issue, so feel free to close it if you want.

I have a dataset with not-labelled call transcriptions, and I want to train a classifier for them. I'm wondering if I could use the few-shot-ssl-public to train it (once part of the dataset it's labelled manually).

I'm waiting your suggestion, thanks again!

**Comment by renmengye** [comment]
Yes, I think you can use this to train your classifier to learn a semi-supervised algorithm, assuming that your dataset is also few-shot based.

**Comment by iuria21** [comment]
thanks for answering. Sorry for my ignorance, when you say "assuming that your dataset is also few-shot based", what do you mean? I mean, I want to train the model from scratch, I have not a trained model and want to add a new label with few examples.

**Comment by iuria21** [comment]
and, in what format should the data be fed to the network? I'm not able to see it

**Comment by renmengye** [comment]
Sorry, so it means you probably have many classes to sample few-shot episodes from.

**Comment by iuria21** [comment]
well, thanks for answering, this is my case. Now I'm making trials with the code, I trained the "omniglot" dataset succesfully. I'm finding the way to introduce my own data, I have it in numpy arrays or lists of (Number_examples, Max_length, Feature_dim) which could be parallel to the image dataset types. What I don't know is how should be it pickled in order to train a model with it. I suppouse that I have to pickle my data as in your examples, but I don't know how the labels are encoded not what is the label_str parameter. Could you help me with that?
**EDIT**:
I pickled my dataset in the format with images, labels and label_str. Now I can get the model, but I changed the height and the width in the config file and I'm getting this error (mine is 30x50 instead of 28x28):
```
InvalidArgumentError (see above for traceback): Incompatible shapes: [1,4,192] vs. [1,5,1]
	 [[node MetaTrain/Model/mul (defined at /few-shot-ssl-public/fewshot/models/basic_model.py:93)  = Mul[T=DT_FLOAT, _device="/job:localhost/replica:0/task:0/device:CPU:0"](MetaTrain/Model/split, MetaTrain/Model/ExpandDims)]]
```
